### PR TITLE
Build out Contribute index with download instructions and descriptions of primary directories

### PR DESCRIPTION
**Pull request by Clay-Mysten**
Move `git clone` command to Contribute in prep for removal from tutorial
Add link to Install docs for getting binaries

### CODE REVIEW

**Comment by Clay-Mysten** [comment]
Hi team,

Adam had the excellent idea of summarizing our primary directories in our Contribute page. I have taken a first step but would like your input. Feel free to commit changes directly or provide them in comments. And of course add missing reviewers. Thanks!

File: doc/src/contribute/index.md [hunk 1]
```diff
@@ -16,6 +18,23 @@ You may also suggest changes to the docs directly in GitHub right here using the
 
 Simply edit the file in question and generate a pull request. We will get back to you shortly.
 
+## Download and learn Sui?
```
Comment (awelc) [inline doc/src/contribute/index.md:21]:
```suggestion
## Download and learn Sui
```

File: doc/src/contribute/index.md [hunk 2]
```diff
@@ -16,6 +18,23 @@ You may also suggest changes to the docs directly in GitHub right here using the
...
+
+In order to obtain the Sui source code, clone the Sui repository:
+
+```shell
+git clone https://github.com/MystenLabs/sui.git
+```
+
+The primary directories are:
```
Comment (awelc) [inline doc/src/contribute/index.md:29]:
```suggestion
You can start exploring Sui's source code by looking into the following primary directories:
```

It's just a suggestion, though.

File: doc/src/contribute/index.md [hunk 3]
```diff
@@ -16,6 +18,23 @@ You may also suggest changes to the docs directly in GitHub right here using the
...
+
+* [sui](https://github.com/MystenLabs/sui/tree/main/sui) - the Sui binaries (`wallet`, `sui-move`, and more)
+* [sui_programmability](https://github.com/MystenLabs/sui/tree/main/sui_programmability) - games and other examples for testing and reuse
```
Comment (awelc) [inline doc/src/contribute/index.md:32]:
```suggestion
* [sui_programmability](https://github.com/MystenLabs/sui/tree/main/sui_programmability) - Sui's Move language integration also including games and other Move code examples for testing and reuse
```

File: doc/src/contribute/index.md [hunk 4]
```diff
@@ -16,6 +18,23 @@ You may also suggest changes to the docs directly in GitHub right here using the
...
+* [sui_core](https://github.com/MystenLabs/sui/tree/main/sui_core) - authority server and Sui Gateway
+* [sui_types](https://github.com/MystenLabs/sui/tree/main/sui_types) - coins, gas, and other object types
+* [explorer](https://github.com/MystenLabs/sui/tree/main/explorer) - chain explorer for the Sui network
```
Comment (awelc) [inline doc/src/contribute/index.md:35]:
```suggestion
* [explorer](https://github.com/MystenLabs/sui/tree/main/explorer) - object explorer for the Sui network
```

**Comment by awelc** [comment]
> Hi team,
> 
> Adam had the excellent idea of summarizing our primary directories in our Contribute page. I have taken a first step but would like your input. Feel free to commit changes directly or provide them in comments. And of course add missing reviewers. Thanks!

Let me elaborate on this a bit more. The main motivation for this whole change is to balance two things out:
1. We don't necessarily want people that just want to play with Sui via it's CLI (`sui`, `wallet`, etc.) to be forced to clone the entire repo as it's not really useful much for this kind of audience
2. At the same time, in some docs we are referring to code in Sui's repo and ask people to "download" it somehow (e.g., Move code to be published for playing the TicTacToe game). In these cases it would be convenient to point people at some common location where we describe how to clone our repo.

The contribute section seems like a decent place to put repo cloning instructions as this file targets folks that actually need to inspect Sui's source code. Having bare-bones cloning instructions could be OK, but it just feels better to also include a very brief code structure blurb there as well.

**Review by awelc — APPROVED** [review]
Stamped, but since this was already something that @Clay-Mysten  and I discussed, it would be good to get at least one more opinion on this from someone that is not biased.

**Pull request merged**